src/main/java/com/dnsimple/request/CertificateRenewalPurchaseOptions.java

```diff
@@ -0,0 +1,32 @@
+package com.dnsimple.request;
+
+public class CertificateRenewalPurchaseOptions {
+    private final boolean autoRenew;
+    private final SignatureAlgorithm signatureAlgorithm;
+
+    private CertificateRenewalPurchaseOptions(boolean autoRenew, SignatureAlgorithm signatureAlgorithm) {
+        this.autoRenew = autoRenew;
+        this.signatureAlgorithm = signatureAlgorithm;
+    }
+
+    /**
+     * @param name The name of the certificate to be renewed
+     */
+    public static CertificateRenewalPurchaseOptions of(String name) {
+        return new CertificateRenewalPurchaseOptions(false, SignatureAlgorithm.ECDSA);
+    }
```

Comment:
Yes this is a good point, the `of` static method was copied but should not have been kept. I'll use the more straightforward `empty()` method for now to avoid the more involved changes. Going forward, WDYT about using a standard builder pattern for these data classes with optional fields, instead of what we currently do? I feel like it could provide a better experience for library users and contributors, but I may be missing something as I'm not sure about the context for the current design decisions.

- As your involved changes describe, we currently have nonstandard ways of constructing these classes that could change, whereas the builder pattern is much more standard and Java users and tools will be able to understand them immediately. For example, it isn't clear why `of(String name)` or `rsa()`/`esdsa()` specifically should be the entry point, as it's not the "main" argument nor the only required one, and even if it is for this class it's not generally applicable to all data classes, causing an inconsistent experience. Whereas with something like `new Options().arg1(true).arg2("www").build()`, it's universally applicable and usable to all data-like objects. This is currently how it's implemented in our Python, Rust, Node.js, and Go libraries, as well as some other popular SDKs like AWS, Azure, and GCP.
- I think we should avoid hard coding default values for any arguments (including optional ones) and omit their serialisation, so that the library does not go out of sync with the API server which should be the source of truth, which could cause bugs due to misleading assumptions. This is also currently how it's implemented in our other libraries.

Reply:
Going with `empty()` at this point sounds good 👍 

Thanks for sharing your insights too, @wilson-dnsimple! I think you are making fair points, and going for a builder pattern would be a nice choice here. I'd love to see that happening :) ❤️